### PR TITLE
Chore: Add a .gitignore

### PR DESCRIPTION
**Pull request by LudwigThePig**
Was there a reason to omit a .gitignore?

This .gitignore file is the basic node boilerplate that GitHub provides. Let me know if you want to go a little more lean, like just ignore node_modules and os level stuff.

Cheers,
Morgan

### CODE REVIEW

**Comment by johakr** [comment]
> Yeah pushing `node_modules` is deliberate. The dependency list is extremely small and only for development. I'll leave this open since I'm sure this question will come up more.

https://github.com/facebookexperimental/rome/issues/26#issuecomment-591787339

**Comment by Luchanso** [comment]
Oh, yes, this is need for Continuous Integration.

**Comment by TamalAnwar** [comment]
Hi @LudwigThePig The node_modules folder is included because the Rome project is everything we will need to do testing, compiling and stuff. The node_module folder now holds the typescript files, but maybe in the future they will be merged to the packages folder.

**Comment by LudwigThePig** [comment]
@TamalAnwar @johakr Ahh, I see. So you want people to be able to clone Rome and start running, without running `npm install` or `yarn install`

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+bower_components
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
```
Comment (ericlewis) [inline .gitignore]:
```suggestion

```

Comment (LudwigThePig) [reply .gitignore]: [Fixed per commit](https://github.com/facebookexperimental/rome/pull/64/commits/fafe75fac50f0bcf00f885b425d68115c1344a2f)

Also,
- removed a lot of extraneous rules
- added rules specific to IDEs/text editors ([I stole them from the React .gitignore](https://github.com/facebook/react/blob/master/.gitignore))

**Comment by sebmck** [comment]
There's no value in a `.gitignore` because there's no files to ignore. `node_modules` is meant to be included, see #26. Including things like editor directories doesn't feel necessary either since it's pushing specific developer assumptions into the repo. You can use a global git ignore for that.

**Comment by zinyando** [comment]
@sebmck what about in the `@romejs-web` package, a `.gitignore` is surely needed there right?

**Comment by sebmck** [comment]
The web UI package shouldn't have any development done on it because there's a lot of things that need to be done before, including self hosting basic package fetching that wouldn't even require a `node_modules` or another package manager.

**Pull request closed without merging**